src/mesh/caching-mesh-adapter-decorator.ts

```diff
@@ -28,8 +28,15 @@ export class CachingMeshAdapterDecorator implements MeshAdapter {
       );
       const tenants = await this.meshAdapter.getMeshTenants();
 
-      // Transfer cost data from cached repo tenants to new collected tenants.
+      const cachedTenantsById = await this.getCachedTenantsMappedById();
+
+      // Transfer cost & IAM data from cached repo tenants to new collected tenants.
       for (const t of tenants) {
+        const ct = cachedTenantsById.get(t.platformTenantId);
```

Comment:
interesting that I missed that. I actually saw it but somehow I had an answer in my head how you managed to re-attach the cost info onto the cached tenants.

Reply:
Yeah I was also somehow very confident that this was not needed anymore 🤷